Build failure: sane-frontends

## Steps To Reproduce

Steps to reproduce the behavior:

1. build *sane-frontends*

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
this derivation will be built:
  /nix/store/299fkgp23fwkaipkxpl7sddqadx7ml0h-sane-frontends-1.0.14.drv
building '/nix/store/299fkgp23fwkaipkxpl7sddqadx7ml0h-sane-frontends-1.0.14.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/75rsjx8jqk49p7j6sirhshw3h93a6cpz-sane-frontends-1.0.14.tar.gz
source root is sane-frontends-1.0.14
setting SOURCE_DATE_EPOCH to timestamp 1127045414 of file sane-frontends-1.0.14/Changelog
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: configurePhase
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
configure flags: --prefix=/nix/store/8mhsyv369dg20kzcsiisbb6xljnbjdwq-sane-frontends-1.0.14
checking whether GIMP plugin is requested... yes, GIMP 2.0
checking for gcc... gcc
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables...
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ANSI C... none needed
checking how to run the C preprocessor... gcc -E
checking for egrep... grep -E
checking for AIX... no
checking for ANSI C header files... no
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking for library containing strerror... none required
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ANSI C... (cached) none needed
checking for a BSD-compatible install... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/install -c
checking whether make sets $(MAKE)... yes
checking how to run the C preprocessor... gcc -E
checking whether gcc needs -traditional... no
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for ANSI C header files... (cached) no
checking os2.h usability... no
checking os2.h presence... no
checking for os2.h... no
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking for unistd.h... (cached) yes
checking libc.h usability... no
checking libc.h presence... no
checking for libc.h... no
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking for sys/types.h... (cached) yes
checking for an ANSI C-conforming const... yes
checking for inline... inline
checking return type of signal handlers... void
checking for size_t... yes
checking for pid_t... yes
checking for ssize_t... yes
checking for u_char... yes
checking for u_int... yes
checking for u_long... yes
checking for u_int8_t... yes
checking for u_int16_t... yes
checking for u_int32_t... yes
checking for sqrt in -lm... yes
checking for socket in -lsocket... no
checking for syslog in -lsyslog... no
checking for working alloca.h... yes
checking for alloca... yes
checking for stdlib.h... (cached) yes
checking for unistd.h... (cached) yes
checking for getpagesize... yes
checking for working mmap... yes
checking for atexit... yes
checking for mkdir... yes
checking for sigprocmask... yes
checking for strdup... yes
checking for strndup... yes
checking for strftime... yes
checking for strstr... yes
checking for strsep... yes
checking for strtod... yes
checking for snprintf... yes
checking for usleep... yes
checking for strcasecmp... yes
checking for strncasecmp... yes
checking for getenv... yes
checking for isfdtype... yes
checking for vsyslog... yes
checking for ranlib... ranlib
checking for sane-config... /nix/store/1q67cnppn7dnksadpkg93c4kqhq7v9b3-sane-backends-1.3.1/bin/sane-config
checking for SANE - version >= 1.0.0... no
*** Could not run SANE test program, checking why...
*** The test program compiled, but did not run. This usually means
*** that the run-time linker is not finding SANE or finding the wrong
*** version of SANE. If it is not finding SANE, you'll need to set your
*** LD_LIBRARY_PATH environment variable, or edit /etc/ld.so.conf to point
*** to the installed location.  Also, make sure you have run ldconfig if that
*** is required on your system.
***
*** If you have an old version installed, it is best to remove it, although
*** you may also be able to get things to work by modifying LD_LIBRARY_PATH
***
**********************************************************************
ERROR: Couldn't find SANE libraries (sane-backends). Possible reasons:
     - sane-backends isn't installed (install sane-backends before
       sane-frontends)
     - the SANE header files aren't installed (if you installed
       SANE as RPM make sure you also included the sane-devel RPM)
     - the SANE libraries can't be found because /usr/local/lib/ isn't
       searched by the dynamic linker (see INSTALL for details)
**********************************************************************

Running phase: buildPhase
no Makefile or custom buildPhase, doing nothing
Running phase: glibPreInstallPhase
Running phase: installPhase
no Makefile or custom installPhase, doing nothing
Running phase: dropIconThemeCache
Running phase: glibPreFixupPhase
Running phase: fixupPhase
error: builder for '/nix/store/299fkgp23fwkaipkxpl7sddqadx7ml0h-sane-frontends-1.0.14.drv' failed to produce output path for output 'out' at '/nix/store/299fkgp23fwkaipkxpl7sddqadx7ml0h-sane-frontends-1.0.14.drv.chroot/root/nix/store/8mhsyv369dg20kzcsiisbb6xljnbjdwq-sane-frontends-1.0.14'
```

<summary>Build Log with patch</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/75rsjx8jqk49p7j6sirhshw3h93a6cpz-sane-frontends-1.0.14.tar.gz
source root is sane-frontends-1.0.14
setting SOURCE_DATE_EPOCH to timestamp 1127045414 of file sane-frontends-1.0.14/Changelog
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/imrpay9wkywlx5ckbfnrv1inq3cgv3dq-patch.diff
patching file acinclude.m4
patching file aclocal.m4
patching file configure
patching file configure.in
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
configure flags: --prefix=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14
checking whether GIMP plugin is requested... yes, GIMP 2.0
checking for gcc... gcc
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables... 
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ANSI C... none needed
checking how to run the C preprocessor... gcc -E
checking for egrep... grep -E
checking for AIX... no
checking for ANSI C header files... no
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking for library containing strerror... none required
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ANSI C... (cached) none needed
checking for a BSD-compatible install... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/install -c
checking whether make sets $(MAKE)... yes
checking how to run the C preprocessor... gcc -E
checking whether gcc needs -traditional... no
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking for ANSI C header files... (cached) no
checking os2.h usability... no
checking os2.h presence... no
checking for os2.h... no
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking for unistd.h... (cached) yes
checking libc.h usability... no
checking libc.h presence... no
checking for libc.h... no
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking for sys/types.h... (cached) yes
checking for an ANSI C-conforming const... yes
checking for inline... inline
checking return type of signal handlers... void
checking for size_t... yes
checking for pid_t... yes
checking for ssize_t... yes
checking for u_char... yes
checking for u_int... yes
checking for u_long... yes
checking for u_int8_t... yes
checking for u_int16_t... yes
checking for u_int32_t... yes
checking for sqrt in -lm... yes
checking for socket in -lsocket... no
checking for syslog in -lsyslog... no
checking for working alloca.h... yes
checking for alloca... yes
checking for stdlib.h... (cached) yes
checking for unistd.h... (cached) yes
checking for getpagesize... yes
checking for working mmap... yes
checking for atexit... yes
checking for mkdir... yes
checking for sigprocmask... yes
checking for strdup... yes
checking for strndup... yes
checking for strftime... yes
checking for strstr... yes
checking for strsep... yes
checking for strtod... yes
checking for snprintf... yes
checking for usleep... yes
checking for strcasecmp... yes
checking for strncasecmp... yes
checking for getenv... yes
checking for isfdtype... yes
checking for vsyslog... yes
checking for ranlib... ranlib
checking for sane-config... /nix/store/1q67cnppn7dnksadpkg93c4kqhq7v9b3-sane-backends-1.3.1/bin/sane-config
checking for SANE - version >= 1.0.0... yes
checking for pkg-config... /nix/store/0d4m43yp69lrm8imxbqgl9zxjwwz52jw-pkg-config-wrapper-0.29.2/bin/pkg-config
checking for GTK+ - version >= 2.0.0... yes (version 2.24.33)
checking for pkg-config... (cached) /nix/store/0d4m43yp69lrm8imxbqgl9zxjwwz52jw-pkg-config-wrapper-0.29.2/bin/pkg-config
checking for GIMP - version >= 1.3.23... Package gimp-2.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gimp-2.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'gimp-2.0' found
Package gimpui-2.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gimpui-2.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'gimpui-2.0' found
*** Could not run GIMP test program, checking why...
*** The test program failed to compile or link. See the file config.log for the
*** exact error that occured. This usually means GIMP is incorrectly installed.
configure: creating ./config.status
config.status: creating Makefile
config.status: creating lib/Makefile
config.status: creating src/Makefile
config.status: creating sanei/Makefile
config.status: creating doc/Makefile
config.status: creating include/Makefile
config.status: creating include/sane/config.h
****************************************************************
*                                                              *
* SANE-FRONTENDS configure status:                             *
* --------------------------------                             *
* - GUIs         activated (xcam + xscanimage)                 *
* - GIMP plugin  deactivated for xscanimage                    *
*                                                              *
****************************************************************
*                                                              *
* To compile SANE-FRONTENDS:                                   *
* --------------------------                                   *
*   make                                                       *
*   make install                                               *
*                                                              *
* ------------------------------------------------------------ *
* ... PLEASE READ SANE DOCUMENTATION BEFORE STARTING SANE ...  *
* ------------------------------------------------------------ *
*                                                              *
* If you experience any problems with sane-frontends please    *
* write to the sane mailing list or contact:                   *
* Oliver Rauch <Oliver.Rauch@Rauch-domain.de>                  *
*                                                              *
****************************************************************
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j8 SHELL=/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/bash
making all in include
make[1]: Entering directory '/build/sane-frontends-1.0.14/include'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/build/sane-frontends-1.0.14/include'
making all in lib
make[1]: Entering directory '/build/sane-frontends-1.0.14/lib'
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include/sane -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall alloca.c
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include/sane -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall getopt.c
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include/sane -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall getopt1.c
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include/sane -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall strndup.c
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include/sane -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall strdup.c
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include/sane -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall strsep.c
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include/sane -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall snprintf.c
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include/sane -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall usleep.c
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include/sane -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall isfdtype.c
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include/sane -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall vsyslog.c
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include/sane -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall getenv.c
ar r liblib.a alloca.o getopt.o getopt1.o strndup.o strdup.o strsep.o snprintf.o usleep.o isfdtype.o vsyslog.o getenv.o
ar: creating liblib.a
ranlib liblib.a
make[1]: Leaving directory '/build/sane-frontends-1.0.14/lib'
making all in sanei
make[1]: Entering directory '/build/sane-frontends-1.0.14/sanei'
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall -I/nix/store/1q67cnppn7dnksadpkg93c4kqhq7v9b3-sane-backends-1.3.1/include sanei_init_debug.c
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall -I/nix/store/1q67cnppn7dnksadpkg93c4kqhq7v9b3-sane-backends-1.3.1/include sanei_wire.c
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall -I/nix/store/1q67cnppn7dnksadpkg93c4kqhq7v9b3-sane-backends-1.3.1/include sanei_codec_ascii.c
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall -I/nix/store/1q67cnppn7dnksadpkg93c4kqhq7v9b3-sane-backends-1.3.1/include sanei_codec_bin.c
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall -I/nix/store/1q67cnppn7dnksadpkg93c4kqhq7v9b3-sane-backends-1.3.1/include sanei_save_values.c
gcc -c -DHAVE_CONFIG_H -I. -I. -I../include -I../include -D_GNU_SOURCE 	  -DPATH_SANE_DATA_DIR=/nix/store/720i6sv0kp2fv77h7jmri51nzzpiw42p-sane-frontends-1.0.14/share/sane 	  -DV_MAJOR=1 -DV_MINOR=0 -DV_BUILD=14 -DSANE_V_MAJOR=1 -g -O2 -Wall -I/nix/store/1q67cnppn7dnksadpkg93c4kqhq7v9b3-sane-backends-1.3.1/include sanei_load_values.c
[01m[Ksanei_wire.c:[m[K In function '[01m[Ksanei_w_array[m[K':
[01m[Ksanei_wire.c:213:23:[m[K [01;35m[Kwarning: [m[Kformat '[01m[K%u[m[K' expects argument of type '[01m[Kunsigned int[m[K', but argument 4 has type '[01m[Ksize_t[m[K' {aka '[01m[Klong unsigned int[m[K'} [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-Wformat=]8;;[m[K]
  213 |               DBG (0, [01;35m[K"sanei_w_array: DECODE: maximum amount of allocated memory "[m[K
      |                       [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
  214 |                    "exceeded (limit: %u, new allocation: %u, total: %u bytes)\n",
  215 |                    MAX_MEM, [32m[Klen * element_size[m[K, MAX_MEM + len * element_size);
      |                             [32m[K~~~~~~~~~~~~~~~~~~[m[K
      |                                 [32m[K|[m[K
      |                                 [32m[Ksize_t {aka long unsigned int}[m[K
[01m[Ksanei_wire.c:214:59:[m[K [01;36m[Knote: [m[Kformat string is defined here
  214 |                    "exceeded (limit: %u, new allocation: [01;36m[K%u[m[K, total: %u bytes)\n",
      |                                                          [01;36m[K~^[m[K
      |                                                           [01;36m[K|[m[K
      |                                                           [01;36m[Kunsigned int[m[K
      |                                                          [32m[K%lu[m[K
[01m[Ksanei_wire.c:213:23:[m[K [01;35m[Kwarning: [m[Kformat '[01m[K%u[m[K' expects argument of type '[01m[Kunsigned int[m[K', but argument 5 has type '[01m[Ksize_t[m[K' {aka '[01m[Klong unsigned int[m[K'} [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-Wformat=]8;;[m[K]
  213 |               DBG (0, [01;35m[K"sanei_w_array: DECODE: maximum amount of allocated memory "[m[K
      |                       [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
[01m[Ksanei_wire.c:214:70:[m[K [01;36m[Knote: [m[Kformat string is defined here
  214 |                    "exceeded (limit: %u, new allocation: %u, total: [01;36m[K%u[m[K bytes)\n",
      |                                                                     [01;36m[K~^[m[K
      |                                                                      [01;36m[K|[m[K
      |                                                                      [01;36m[Kunsigned int[m[K
      |                                                                     [32m[K%lu[m[K
[01m[Ksanei_wire.c:[m[K In function '[01m[Ksanei_w_ptr[m[K':
[01m[Ksanei_wire.c:291:23:[m[K [01;35m[Kwarning: [m[Kformat '[01m[K%u[m[K' expects argument of type '[01m[Kunsigned int[m[K', but argument 4 has type '[01m[Ksize_t[m[K' {aka '[01m[Klong unsigned int[m[K'} [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-Wformat=]8;;[m[K]
  291 |               DBG (0, [01;35m[K"sanei_w_ptr: DECODE: maximum amount of allocated memory "[m[K
      |                       [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
  292 |                    "exceeded (limit: %u, new allocation: %u, total: %u bytes)\n",
  293 |                    MAX_MEM, [32m[Kvalue_size[m[K, (w->allocated_memory + value_size));
      |                             [32m[K~~~~~~~~~~[m[K
      |                             [32m[K|[m[K
      |                             [32m[Ksize_t {aka long unsigned int}[m[K
[01m[Ksanei_wire.c:292:59:[m[K [01;36m[Knote: [m[Kformat string is defined here
  292 |                    "exceeded (limit: %u, new allocation: [01;36m[K%u[m[K, total: %u bytes)\n",
      |                                                          [01;36m[K~^[m[K
      |                                                           [01;36m[K|[m[K
      |                                                           [01;36m[Kunsigned int[m[K
      |                                                          [32m[K%lu[m[K
[01m[Ksanei_wire.c:291:23:[m[K [01;35m[Kwarning: [m[Kformat '[01m[K%u[m[K' expects argument of type '[01m[Kunsigned int[m[K', but argument 5 has type '[01m[Ksize_t[m[K' {aka '[01m[Klong unsigned int[m[K'} [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-Wformat=]8;;[m[K]
  291 |               DBG (0, [01;35m[K"sanei_w_ptr: DECODE: maximum amount of allocated memory "[m[K
      |                       [01;35m[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
  292 |                    "exceeded (limit: %u, new allocation: %u, total: %u bytes)\n",
  293 |                    MAX_MEM, value_size, [32m[K(w->allocated_memory + value_size)[m[K);
      |                                         [32m[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[m[K
      |                                                              [32m[K|[m[K
      |                                                              [32m[Ksize_t {aka long unsigned int}[m[K
[01m[Ksanei_wire.c:292:70:[m[K [01;36m[Knote: [m[Kformat string is defined here
  292 |                    "exceeded (limit: %u, new allocation: %u, total: [01;36m[K%u[m[K bytes)\n",
      |                                                                     [01;36m[K~^[m[K
      |                                                                      [01;36m[K|[m[K
      |                                                                      [01;36m[Kunsigned int[m[K
      |                                                                     [32m[K%lu[m[K
[01m[Ksanei_init_debug.c:[m[K In function '[01m[Ksanei_debug_msg[m[K':
[01m[Ksanei_init_debug.c:103:17:[m[K [01;31m[Kerror: [m[Kimplicit declaration of function '[01m[Kisfdtype[m[K'; did you mean '[01m[Kisctype[m[K'? [[01;31m[K]8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration]8;;[m[K]
  103 |       if ( 1 == [01;31m[Kisfdtype[m[K(fileno(stderr), S_IFSOCK) )
      |                 [01;31m[K^~~~~~~~[m[K
      |                 [32m[Kisctype[m[K
make[1]: *** [Makefile:68: sanei_init_debug.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/build/sane-frontends-1.0.14/sanei'
make: *** [Makefile:78: all-recursive] Error 1
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

I found a [bug report](https://bugs.gentoo.org/933615) with a [patch](https://bugs.gentoo.org/attachment.cgi?id=903172&action=diff) that at least makes the build start but it still fails.. I did not find the patches mentioned in the bug report, I think they should be [here](https://salsa.debian.org/debian/sane-frontends/-/tree/master/debian/patches?ref_type=heads).. I could not open an issue in gitlab because I can't get around the cloudflare captcha for some reason :(

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@symphorien @melling 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/graphics/sane/frontends.nix
+++ b/pkgs/applications/graphics/sane/frontends.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchurl,
+  fetchpatch,
   sane-backends,
   libX11,
   gtk2,
@@ -18,9 +19,47 @@ stdenv.mkDerivation rec {
     sha256 = "1ad4zr7rcxpda8yzvfkq1rfjgx9nl6lan5a628wvpdbh3fn9v0z7";
   };
 
-  preConfigure = ''
-    sed -e '/SANE_CAP_ALWAYS_SETTABLE/d' -i src/gtkglue.c
-  '';
+  # add all fedora patchs. fix gcc-14 build among other things
+  # https://src.fedoraproject.org/rpms/sane-frontends/tree/main
+  patches =
+    let
+      fetchFedoraPatch =
+        { name, hash }:
+        fetchpatch {
+          inherit name hash;
+          url = "https://src.fedoraproject.org/rpms/sane-frontends/raw/89f752d7e236e86be8d64b7ac6991a36f9e9f7d0/f/${name}";
+        };
+    in
+    map fetchFedoraPatch [
+      {
+        name = "0001-src-scanadf.c-Fix-segfault-when-scanadf-h-d-device.patch";
+        hash = "sha256-sSUWm5fL7YTebzXh3Thb/qwgr7d++1Y+74uI8R5oF0g=";
+      }
+      {
+        name = "frontends-scanadf-segv.patch";
+        hash = "sha256-VRag9nMk8ZCjg9Oq0siHdT8J6sbNjq9cU2ktOH2vkLo=";
+      }
+      {
+        name = "sane-frontends-1.0.14-array-out-of-bounds.patch";
+        hash = "sha256-a0lzbAogSrXsK5jVeNffDS+zFxpuDHXpHQlOJ5874+U=";
+      }
+      {
+        name = "sane-frontends-1.0.14-sane-backends-1.0.20.patch";
+        hash = "sha256-ViYjxXGj58P6EaZ+fIiAydrgbyS1ivn39uN3EWcvnZg=";
+      }
+      {
+        name = "sane-frontends-1.0.14-xcam-man.patch";
+        hash = "sha256-HGANgQPujn/jjOMGs9LlzYvYZphMWwbsI74NCad5ADc=";
+      }
+      {
+        name = "sane-frontends-c99.patch";
+        hash = "sha256-LPELEG11wEom05ECAMgXUDRWvrbuU4nT3apuS1eITyA=";
+      }
+      {
+        name = "sane-frontends-configure-c99.patch";
+        hash = "sha256-SPvMDCZv8VRGP+cXRFjVbqgbTeVhdLOTEQbbBgSMLvY=";
+      }
+    ];
 
   buildInputs = [
     sane-backends

